tests/shouldwork/Basic/T1591.hs

```diff
@@ -32,8 +32,7 @@ data NextLineMask this that =
   } deriving (Generic)
 
 nextLineMask :: forall this that. (KnownNat this, KnownNat that) => Index (CountStates this that) -> NextLineMask this that
-nextLineMask =
-  errorX ""
+nextLineMask x = NextLineMask x
```

Comment:
This change is needed because otherwise we get this failure: https://gitlab.com/clash-lang/clash-compiler/-/jobs/2269702690

Why does this failure pop up you might ask. That's because https://github.com/clash-lang/clash-compiler/pull/2153/files#diff-64d9f9192b0238fa29af4ce5d35293874d555046eb4708c9e1e9f56fa992c617R466-R467 transforms:
```haskell
case errorX "" of {}
```
to
```haskell
Clash.Normalize.Primitives.undefinedX
```
Which are in turn treated like values in the primitive evaluator here https://github.com/clash-lang/clash-compiler/pull/2153/files#diff-6a8afb744be8dbdbd693897804e34547af6fbd7a6e44bbe4a0fb79b028fd1986R169 which then results in needing to calculate the type `ty` in https://github.com/clash-lang/clash-compiler/pull/2153/files#diff-6a8afb744be8dbdbd693897804e34547af6fbd7a6e44bbe4a0fb79b028fd1986R319-R320

The ultimate cause of the error by the way is because we throw away casts.